src/reactviews/pages/ExecutionPlan/executionPlanGraph.tsx

```diff
@@ -0,0 +1,203 @@
+				setFindNodeOptions(executionPlanView.getUniqueElementProperties());
+				setZoomNumber(executionPlanView.getZoomLevel());
+
+				let tempQuery = executionPlanState.executionPlanGraphs[graphIndex].query
+				if (graphIndex != 0) {
+					const firstAlphaIndex = tempQuery.search(/[a-zA-Z]/);
+
+					if (firstAlphaIndex !== -1) {
+						tempQuery = tempQuery.slice(firstAlphaIndex);
+					}
+				}
+				setQuery(tempQuery);
+				setCost(executionPlanView.getTotalRelativeCost());
+			}
+			else {
+				return;
+			}
+		}
+		loadExecutionPlan();
+
+	}, [executionPlanState]);
+
+	const handleCustomZoomInput = async () => {
+		if (executionPlanView) {
+			executionPlanView.setZoomLevel(zoomNumber);
+			setExecutionPlanView(executionPlanView);
+			setZoomNumber(executionPlanView.getZoomLevel());
+		}
+		setCustomZoomClicked(false);
+	};
+
+	const getQueryCostPercentage = () => {
+		const percentage = (cost / executionPlanState!.totalCost!) * 100;
+  		return percentage.toFixed(2) + '%';
+	};
+
+	return (
+		<div id="panelContainer" className={classes.panelContainer}>
+			<div id="planContainer" className={classes.planContainer}>
+				<div id="queryCostContainer" className={classes.queryCostContainer} style={{background:utils.background(executionPlanState!.theme!)}}>
+					{formatString(LocalizedConstants.queryCostHeader, graphIndex + 1, getQueryCostPercentage())}<br />{query}
+				</div>
+				<div id={`queryPlanParent${graphIndex + 1}`} className={classes.queryPlanParent}></div>
+				{customZoomClicked ? (
+					<div id="customZoomInputContainer" className={classes.inputContainer} style={{background:utils.iconBackground(executionPlanState!.theme!)}}>
+						<Input id="customZoomInputBox" type="number" min={1} defaultValue={Math.floor(zoomNumber).toString()} onChange={(e) => setZoomNumber(Number(e.target.value))} style={{ width: '100px', height: '25px', fontSize: '12px' }}/>
+						<Button onClick={handleCustomZoomInput} icon={<Checkmark20Regular />} />
+						<Button icon={<Dismiss20Regular />} onClick={() => setCustomZoomClicked(false)}/>
+					</div>
+				) : null}
+				{findNodeClicked ? (
+					<FindNode executionPlanView={executionPlanView} setExecutionPlanView={setExecutionPlanView} findNodeOptions={findNodeOptions} setFindNodeClicked={setFindNodeClicked}/>
+				) : null}
+				{highlightOpsClicked ? (
+					<HighlightExpensiveOperations executionPlanView={executionPlanView} setExecutionPlanView={setExecutionPlanView} setHighlightOpsClicked={setHighlightOpsClicked}/>
+				) : null}
+			</div>
+			<IconStack executionPlanView={executionPlanView} setExecutionPlanView={setExecutionPlanView} setZoomNumber={setZoomNumber} setCustomZoomClicked={setCustomZoomClicked} setFindNodeClicked={setFindNodeClicked} setHighlightOpsClicked={setHighlightOpsClicked} query={query}/>
+		</div>
+	);
+};
+
+/**
+ * Format a string. Behaves like C#'s string.Format() function.
+ */
+function formatString(str: string, ...args: any[]): string {

```

Comment:
should we put this in `utils.ts` to it can be used elsewhere as needed?

Reply:
I moved it to src\reactviews\common\utils.ts